### PR TITLE
feat: add better docs for timestamp.to_{hour,day}

### PR DESCRIPTION
**Pull request by alxu-sentry**
This isn't *quite* ready to land because as @Zylphrex pointed out, filtering on `timestamp.to_hour` doesn't work at the moment. I think we can still review this content now though.

### CODE REVIEW

File: src/docs/product/sentry-basics/guides/discover/index.mdx
Comment (dashed) [inline src/docs/product/sentry-basics/guides/discover/index.mdx:108]:
```suggestion
The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC, will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to roll up events. For example, you can show hourly hit counts for errors hit at least twice per hour:
```

Comment (alxu-sentry) [reply src/docs/product/sentry-basics/guides/discover/index.mdx]: applied change sans comma. thank you for the suggestion.

File: src/docs/product/sentry-basics/guides/discover/index.mdx
Comment (Zylphrex) [inline src/docs/product/sentry-basics/guides/discover/index.mdx:108]: This shouldn't just always use UTC time. Like the `timestamp` field, this should respect the user's timezone preference.

Comment (PeloWriter) [inline src/docs/product/sentry-basics/guides/discover/index.mdx:108]:
```suggestion
The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to group events. For example, you can show hourly hit counts for errors hit at least twice per hour:
```

**Review by PeloWriter — APPROVED** [review]

File: src/docs/product/sentry-basics/guides/discover/index.mdx
Comment (alxu-sentry) [reply src/docs/product/sentry-basics/guides/discover/index.mdx]: There's two parts here: calculation and formatting. For the calculation part, it's complicated to pass through the right time zone value, and raises questions about sharing queries. For the formatting part, that was broken on prod due to setting timezone to `Universal` instead of `UTC`, and was fixed in getsentry/sentry#24443.

**Review by dashed — APPROVED** [review]

**Comment by imatwawana** [comment]
@Zylphrex is the `timestamp.to_hour` filter working or does we still need to hold off on merging this?

**Comment by Zylphrex** [comment]
> @Zylphrex is the `timestamp.to_hour` filter working or do we still need to hold off on merging this?

The filtering is working as expected now. One change I'd like to suggest before merging this is that these fields are not always formatted in UTC. Instead, it respects the user's timezone preference. If we can work that into the docs, then we can merge this 😄

File: src/docs/product/sentry-basics/guides/discover/index.mdx
Comment (imatwawana) [inline src/docs/product/sentry-basics/guides/discover/index.mdx:112]:
```suggestion
All timestamps are displayed in your preferred timezone based on your user settings in **User Settings > Account > Account Details**.
```

**Review by imatwawana — APPROVED** [review]
Made a couple small wording changes for concision and clarity

**Pull request merged**
